### ISSUE TITLE
[$250] Expanding text expander snippets in the composer replace everything typed

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#39366]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @twisterdotcom 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1711714936667039

## Action Performed:
1. Open any report
2. Type some text and add emoji manually or delete some characters and add emoji

## Expected Result:
Text typed before adding emoji should stay

## Actual Result:
Replaces everything typed

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/e41bdc52-f9ef-4198-ac85-0c511b8f85d9

https://github.com/Expensify/App/assets/38435837/d302de8f-b3f2-4273-9a8e-48b094afc461

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01bb81fb54a19672c0</li>
        <li>Upwork Job ID: 1775932365632151552</li>
        <li>Last Price Increase: 2024-04-04</li>
    </ul>
</details>

**Comment by mallenexpensify** [comment]
Snaggin' this from ya as BZ @kevinksullivan .  I experience this all day so wanna help to try to get it fixed quickly. 
@alitoshmatov can you see if you can find a way to reproduce **without** using TextExpander (TE) ?  Maybe you and contributors can download a free version from https://textexpander.com/    Or.. use another snippet/shortcut/expander app.   I don't think I've found a way to repro without TE. 

@twisterdotcom , any new deets from your end since you reported this?

**Comment by twisterdotcom** [comment]
No new deets from me. This just happens with all TE snippets. I tried disabling TextExpander and seeing if the same behaviour happened with Espanso, but it doesn't:

https://github.com/Expensify/App/assets/9133401/8c58c152-80d0-4ccc-b1fc-22080677a2ce

Although, as if by magic, I think we deployed something yesterday which fixed this because this is TextExpander now:

https://github.com/Expensify/App/assets/9133401/8448ef2c-232c-48cc-8d78-4ab22b674a46

@mallenexpensify if you confirm, let's close.
